docs/how-tos/provision-setup.md

```diff
@@ -0,0 +1,287 @@
+}
+```
+
+{{% /tablestep %}}
+{{< /table >}}
+
+## Install `viam-agent`
+
+`viam-agent` is a self-updating service manager that maintains the lifecycle for several Viam services and keeps them updated.
+If you intend to use `viam-agent` to keep your device's Viam software up-to-date or to use its provisioning plugin to let end users set up their own machines, you need to use `viam-agent` and install it on the machine before sending it to the user.
+
+You can [install `viam-agent` on live systems](/installation/) or preinstall it:
+
+{{< tabs >}}
+{{% tab name="Preinstall on an SD card (or other image)" %}}
+
+Viam provides a preinstall script that works with images to add `viam-agent`.
+
+{{< alert title="Support notice" color="note" >}}
+Please note this script works only under POSIX (MacOS and Linux) at the moment.
+{{< /alert >}}
+
+{{< table >}}
+{{< tablestep >}}
+**1. Download the preinstall script**
+
+Run the following commands to download the preinstall script and make the script executable:
+
+```sh {class="command-line" data-prompt="$"}
+wget https://storage.googleapis.com/packages.viam.com/apps/viam-agent/preinstall.sh
+chmod 755 preinstall.sh
+```
+
+{{< /tablestep >}}
+{{< tablestep >}}
+**2. Run the preinstall script**
+
+Run the preinstall script without options and it will attempt to auto-detect a mounted root filesystem (or for Raspberry Pi, bootfs) and also automatically determine the architecture.
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh
+```
+
+Follow the instructions.
+If you created a <FILE>viam-provisioning.json</FILE>, specify its location when prompted.
+
+```sh {class="command-line" data-prompt="$" data-output="2-40"}
+sudo ./preinstall.sh
+
+
+Found Raspberry Pi bootfs mounted at /Volumes/bootfs
+
+
+A Raspberry Pi boot partition has been found mounted at /Volumes/bootfs
+This script will modify firstrun.sh on that partition to install Viam agent.
+Continue pre-install? (y/n): y
+Path to custom viam-provisioning.json (leave empty to skip):
+Creating tarball for install.
+a opt
+a opt/viam
+a opt/viam/cache
+a opt/viam/bin
+a opt/viam/bin/viam-agent
+a opt/viam/bin/agent-provisioning
+a opt/viam/cache/viam-agent-provisioning-factory-aarch64
+a opt/viam/cache/viam-agent-factory-aarch64
+a etc
+a usr
+a usr/local
+a usr/local/lib
+a usr/local/lib/systemd
+a usr/local/lib/systemd/system
+a usr/local/lib/systemd/system/viam-agent.service
+a usr/local/lib/systemd/system/multi-user.target.wants
+a usr/local/lib/systemd/system/multi-user.target.wants/viam-agent.service
+
+
+Install complete! You can eject/unmount and boot the image now.
+```
+
+If you get an error, run the script for the target system's architecture:
+
+{{< tabs >}}
+{{% tab name="arm64" %}}
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh --aarch64
+```
+
+{{% /tab %}}
+{{% tab name="x86_64" %}}
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh --x86_64
+```
+
+{{% /tab %}}
+{{< /tabs >}}
```

Comment:
These commands need more elaboration. They create a tarball the user needs to then manually extract into their image/filesystem as a quicker way to do a manual install. They're not just what to run if there is an error.

Reply:
I've added more info based on reading the script. I don't know a lot about mounting file systems so please add info if needed